contrib/tvmop/compile.py

```diff
@@ -37,23 +40,57 @@ def get_target(device):
     parser = argparse.ArgumentParser(description="Generate tvm operators")
     parser.add_argument("-o", action="store", required=True, dest="target_path",
                         help="Target path which stores compiled library")
+    parser.add_argument("--config", action="store", required=True, dest="config_path",
+                        help="Path which stores the config file")
     arguments = parser.parse_args()
 
     func_list_llvm = []
     func_list_cuda = []
+    config_spaces = ConfigSpaces()
 
     # TODO: attach instruction features to the library, e.g., avx-512, etc.
-    for operator_def in __OP_DEF__:
-        for sch, args, name in operator_def.invoke_all():
-            if tvm.module.enabled(get_target(operator_def.target)):
-                func_list = func_list_llvm if operator_def.target == "cpu" else func_list_cuda
-                func_lower = tvm.lower(sch, args,
-                                       name=name,
-                                       binds=operator_def.get_binds(args))
-                func_list.append(func_lower)
+    for op in __OP_DEF__:
+        if tvm.module.enabled(get_target(op.target)):
+            func_list = func_list_llvm if op.target == "cpu" else func_list_cuda
+            for each_kwargs in op.arg_combination:
+                if (op.attrs_valid(**each_kwargs)):
+                    name = op.name \
+                        + ''.join(["{}_{}".format(key, each_kwargs[key]) for key in op.attrs])
+                    if op.dispatch is True:
+                        config_space = autotvm.ConfigSpace()
+                        with autotvm.task.ApplyConfig(config_space):
+                            sch, args = op.func(fallback=False, **each_kwargs)
```

Comment:
This requires `fallback` as a mandatory parameter in `op.func`, which is not ideal in terms of usability in my opinion. We should support compiling whatever users define and treat the `fallback` knob as an advanced feature for performance tuning.

A way to achieve such purpose is inspect the signature of `op.func` for keyword `fallback`. If the keyword does not exist, we just compile the op using the default schedule, e.g.
```python
if 'fallback' in str(inspect.signature(op.func)):
    sch, args = op.func(fallback=False, **each_kwargs)
else:
    sch, args = op.func(**each_kwargs)
```
@yzhliu What do you think?

Reply:
Done. I set `self.dispatchable = 'fallback' in inspect.signature(self.func).parameters` in `opdef.py`